lib/workers/bin/worker.rb

```diff
@@ -1,2 +1,10 @@
 type = ARGV.shift
+begin
+  # Remove the ::Runner class which is in all of them.
+  worker_name = type.gsub(/::Runner$/, '')
+  # Will set the process' name to eg. 'EVM Worker: MiqPriorityWorker'
+  # in ps aux, htop, ...
+  Process.setproctitle("EVM Worker: #{worker_name}")
```

Comment:
I believe we should strive to make it work with utilities like smem out of the box, but should not limit the functionality we might require.  I patch smem with sed (`sed -i s/\.27s/\.100s/g /usr/bin/smem`) to show longer command titles anyway.  

I like the current implementation though would not mind smaller titles and agree that we need to change evm_server.rb and evm_watchdog.rb.  Perhaps simply MiqServer and MiqWatchdog respectively.

Here's what the current implementation looks like on my Miq Master appliance:
```
# smem -c 'pid rss pss uss vss swap name command' | grep "[Rr]uby"
 2936    24152    21916    21832   194628    21024 ruby                     ruby /bin/evm_watchdog.rb
43064   172916   169247   169080   482720    15880 ruby                     MiqUiWorker id: 532, uri: http://127.0.0.1:3000
43083   175372   171703   171536   480420    13236 ruby                     MiqWebServiceWorker id: 534, uri: http://127.0.0.1:4000
43061   209076   205423   205276   528200     2356 ruby                     MiqScheduleWorker id: 531
43036   216948   213319   213176   510912     1760 ruby                     MiqEventHandler id: 524, queue: ems
43033   234136   230002   229744   527924     3620 ruby                     Vmware::InfraManager::EventCatcher id: 523, queue: ems_2
42835   235484   231789   231640   609476     5404 ruby                     /var/www/miq/vmdb/lib/workers/bin/evm_server.rb
43023   265304   261170   260912   563536     7888 ruby                     MiqEmsRefreshCoreWorker id: 521, queue: ems_2
43007   277616   273808   273644   594388     2516 ruby                     Vmware::InfraManager::MetricsCollectorWorker id: 517, queue: vmware
43010   284016   280208   280044   587808     5732 ruby                     Vmware::InfraManager::MetricsCollectorWorker id: 518, queue: vmware
52582   284940   281207   281028   619072        0 ruby                     MiqGenericWorker id: 580, queue: generic
43015   305536   301857   301700   595220     2296 ruby                     MiqEmsMetricsProcessorWorker id: 519, queue: ems_metrics_processor
43058   345116   341400   341240   643588     4332 ruby                     MiqReportingWorker id: 530, queue: reporting
43018   349048   345369   345212   638740     2300 ruby                     MiqEmsMetricsProcessorWorker id: 520, queue: ems_metrics_processor
32096   362056   358348   358184   686732        0 ruby                     MiqPriorityWorker id: 579, queue: generic
43053   365632   361929   361780   663220     5000 ruby                     MiqReportingWorker id: 529, queue: reporting
61250   400944   396750   396488   723072        0 ruby                     MiqGenericWorker id: 575, queue: generic
14786   409516   405808   405644   711844        0 ruby                     MiqPriorityWorker id: 578, queue: generic
58777   821248   816962   816688  1130444        0 ruby                     Vmware::InfraManager::RefreshWorker id: 581, queue: ems_2
43080  1264648  1260507  1260248  1577488     9140 ruby                     MiqVimBrokerWorker id: 533
```

I do think we should keep the Miq prefix so we can easily grep the processes out of command line tool output whenever we need.  That being said I would like to rename workers such as "Vmware::InfraManager::EventCatcher" to something along the lines of "MiqEventCatcherInfraVmware" similar to the older naming we had.  That is more of a personal preference so if the technical feasibility of changing it has a large barrier then it might not be worth the change.

Lastly, I don't think there will be an easy solution to tracking workers which have a level of concurrency (Generic/Priority Workers and a few others) with my current tooling for active monitoring (collectd/influxdb/grafana).  For example, collectd's processes module aggregates GenericWorkers such that I would have to write an expression for every "id:" I expect to occur.  This is not maintainable and probably not feasible for the config file. (collectd.conf)  

I will continue to search for an easy to use solution but at the very least proc titles is a huge win in be able to separate worker types via the command title rather than lumping all ruby processes together.